lib/circuit/CircuitState/CircuitState.ts

```diff
@@ -0,0 +1,70 @@
+import { Policy } from '../Policy/Policy';
+
+export class CircuitState {
+
+  private state: 'open' | 'close' | 'half-open' = 'open';
+  private removeExecutionsTimers: number[] = [];
```

Comment:
The name of this variable is an action, I'd suggest to use subject, just simple `timers` would be enough in this context.